src/System.Memory/src/System/ReadOnlySpan.cs

```diff
@@ -29,8 +29,6 @@ public ReadOnlySpan(T[] array)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
-            if (default(T) == null && array.GetType() != typeof(T[]))
```

Comment:
I cleaned up my environment and reverted the cherrypicked PR I thought I needed. It appears I may not need that change.